app/src/main/java/com/example/mvvmcleanarchitecture/core/base/BaseFragment.kt

```diff
@@ -0,0 +1,65 @@
+package com.example.mvvmcleanarchitecture.core.base
+
+import android.os.Bundle
+import android.view.View
+import android.widget.Toast
+import androidx.annotation.LayoutRes
+import androidx.annotation.StringRes
+import androidx.fragment.app.Fragment
+import androidx.lifecycle.observe
+import androidx.viewbinding.ViewBinding
+
+abstract class BaseFragment<T : BaseViewModel, K : ViewBinding>(@LayoutRes contentLayoutId: Int) :
+    Fragment(contentLayoutId) {
+    protected abstract val viewModel: T
+    protected abstract var binding: K?
```

Comment:
Mam do Ciebie pytanie o viewBinding. Załóżmy że mamy fragment na którym "inflatujemy" layout zawierający frameLayout. Potem używałbym właśnie Frameworku TransitionManagera do podmiany layoutu. Na początku inicjalizowałbym 2 sceny: Scene1 i Scene2 podpięte pod ten widget -> frame layout -> to on bedzie parentem dla naszych scen. I teraz pytanie docelowe: mamy 2 odrebne sceny ale pewne widgety by sie ze soba pokrywaly, ich idki w obu bylyby takie same. I teraz gdy uzywam TransitionManager.go(scene1). to potem robie ViewBinding1.bind(view), setuje widoki przez ten viewBinding1 i potem przez 2. Zastanawialem sie czy da sie zrobic jakis sposob zeby moc to setowac 1 metoda ale tutaj przeszkoda jest viewbinding. Bo przez findByViewId sprawa bylaby prostsza ale mniej bezpieczna.

Reply:
Btw. w module 11 będę pokazywał jak można zaprojektować view binding w bazowym fragmencie. Niestety dynamicznego podmieniania sceny nie wziąłem tam pod uwagę :/

Reply:
Mogę napisać taki przykład i udostępnić Ci jeśli to nie problem i by cię to też zainteresowała. Abstrakcję też napisałem i mógłbyś zerknąć czy to jest głupie czy jednak ok ?